src/cmd/services/m3coordinator/downsample/policy.go

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package downsample
+
+// DownsamplingConfiguration is configuration of what to downsample.
+type DownsamplingConfiguration struct {
+	// NonePolicy is the none downsampling policy.
+	NonePolicy NonePolicy `yaml:"nonePolicy"`
+	// AggregationPolicy is the aggregation downsampling policy.
+	AggregationPolicy AggregationPolicy `yaml:"aggregationPolicy"`
```

Comment:
So does this currently mean that if this is provided the downsampling is effectively achieved by aggregating datapoints within a resolution window?

Reply:
Correct, although you can have multiple policies enabled.